Principal_component_analysis/include/CGAL/linear_least_squares_fitting_circles_2.h

```diff
@@ -111,9 +111,9 @@ linear_least_squares_fitting_2(InputIterator first,
 
   // Translate the 2nd order moment calculated about the origin to
   // the center of mass to get the covariance.
-  covariance[0] += -mass * (c.x() * c.x());
-  covariance[1] += -mass * (c.x() * c.y());
-  covariance[2] += -mass * (c.y() * c.y());
+  covariance[0] -= mass * (c.x() * c.x());
```

Comment:
While you are at it, we also have `CGAL::square()` which may be faster depending on the number type, so here ot could be      `square(c.x())`.